### DISCUSSION TITLE
php bin/console doctrine:schema:update --dump-sql - suggests redundant ALTER

### DISCUSSION THREAD

**Discussion by geoff-maddock** [doctrine/doctrine2#8929] [Support Questions]
I'm looking at using 
```
php bin/console doctrine:schema:update --dump-sql
```

To fix some issues between my schema and database.

Based on the output of that statement, I ran this ALTER:
```
ALTER TABLE asset ADD CONSTRAINT FK_2AF5A5C873649CA FOREIGN KEY (managed_by_id) REFERENCES asset_managed_by (id) ON DELETE SET NULL ON UPDATE CASCADE;
```

And looking at my schema, the FK was added as well as an INDEX to that table (asset).

However, when I re-run the same query, it's still asking me to re-run the same ALTER.   

```
php bin/console doctrine:schema:update --dump-sql | grep managed_by_id
     ALTER TABLE asset ADD CONSTRAINT FK_2AF5A5C873649CA FOREIGN KEY (managed_by_id) REFERENCES asset_managed_by (id);
```

What am I missing?

**Comment by SenseException** [comment]
Can you please also add your config/metadata for the involved entities?

**Comment by geoff-maddock** [comment]
I unearthed the issue for the most part.   

For reference, my metadata was set up as:
```
    /**
     * @ORM\ManyToOne(targetEntity="App\Model\Entity\AssetManagedBy")
     * @ORM\JoinColumns({
     *     @ORM\JoinColumn(name="managed_by_id", referencedColumnName="id")
     * })
     */
    private ?AssetManagedBy $managedBy = null;
```

First was that my entity metadata did not have onDelete="SET NULL".   I corrected that.
Second is that there is no way to set onUpdate="CASCADE".   After discussion, I determined this was unnecessary here, so I decided the default was fine. 
In terms of the alignment of the schema's functionality, it works fine now.

The only remaining issue seems to be that doctrine still tries to enforce the FK contraint name "FK_2AF5A5C873649CA".
I would prefer to name the constraint something more human-readable, but the schema update keeps recommending this value.

Is there anywhere that documents the reason for enforcing a constraint name of this format?

**Reply by geoff-maddock** [reply]
Or is there documentation on how the constraint name is generated?

**Comment by SenseException** [comment]
Hm, I can't reproduce this behaviour. The only difference is that my entities are all in `App\Entity\*`. Maybe it can help if you share ORM, DBAL and PHP version too. Also how the mapping of `AssetManagedBy` looks like.
